File: array/2190_most_frequent_number_following_key_in_an_array/2190_most_frequent_number_following_key_in_an_array.py
class Solution:
    def mostFrequent(self, nums: List[int], key: int) -> int:
        l = len(nums)
        result = {}
        for idx, n in enumerate(nums):
            if n == key:
                if idx + 1 >= len(nums):
                    continue
                if nums[idx+1] in result:
                    result[nums[idx+1]] += 1
                else:
                    result[nums[idx+1]] = 1
        i = 0
        m = 0
        for k,v in result.items():
            if v > m:
                m = v
                i = k
        return i
            

File: array/2190_most_frequent_number_following_key_in_an_array/2190_most_frequent_number_following_key_in_an_array.go
package main

func mostFrequent(nums []int, key int) (result int) {
	l := len(nums)
	m := make(map[int]int)
	for i, n := range nums {
		if n == key {
			if i+1 < l {
				m[nums[i+1]] += 1
			}
		}
	}
	//fmt.Println(m)
	max := 0
	for k, v := range m {
		if v > max {
			result = k
			max = v
		}
	}

	return result
}
